lib/progress.ts

```diff
@@ -84,3 +84,51 @@ export interface IPublishProgress {
    */
   abort(): void;
 }
+
+class GlobalOutputHandler {
+  private progressListener: IPublishProgressListener | undefined;
+  private completionProgress: number;
+
+  constructor(completionProgress: number = 0, progressListener?: IPublishProgressListener) {
+    this.progressListener = progressListener;
+    this.completionProgress = completionProgress;
+  }
+
+  public setListener(listener: IPublishProgressListener) {
+    this.progressListener = listener;
+  }
+
+  public setCompletionProgress(progress: number) {
+    this.completionProgress = progress;
+  }
+
+  public publishEvent(eventType: EventType = EventType.DEBUG, text: string) {
+    const progressEvent: IPublishProgress = {
+      message: text,
+      abort: () => {},
```

Comment:
This is bizarre. I think it might be functionally ok but signals a modeling problem IMO.